### ISSUE TITLE
Technical Questions

### ISSUE THREAD

**Issue by JamesMcCullochDickens** [kennymckormick/pyskl#82]
Hello, thanks so much for your work.

I have a few questions.
1. For two person training, are you using the same approach as CTR-GCN where the tensor data for person 2 in a single person action is just the repeated tensor (or zero padding?). Have you tried both the zero padding approach and the duplicate approach (on ntu 120 in particular)?
2. I notice there are more samples after pre-processing in your download links than in the resulting data from the pre-processing script in CTR-GCN. It seems that in CTR-GCN they discarded more training/testing samples, do you have any thoughts on what the difference is? I have carefully gone over the three pre-processing scripts get_raw_denoised_data, get_raw_skes_data, and seq_transformation. Also, did you center the starting frame according to the position of actor 1?
3. For distributed training, are you using sync batch normalization? 
4. In your training loop, have you set torch.backends.cudnn.deterministic = False or True? I notice a huge speedup when its set to False in my local experiments, so if there is no performance cost I'd like to keep it as False. 
5. For the Cosine Annealing learning rate scheduler, what does the by_epoch=False argument do? I do not see this argument in the PyTorch docs, maybe an older version of PyTorch had a different set of input arguments?

Again, thanks for your work.

**Comment by kennymckormick** [comment]
1. We follow CTR-GCN and use zero-padding. Zero padding is better than duplicate. 
2. I'm not sure. In my pre-processing script I do not discard any training / testing samples (We should not do that). I do center the starting frame according to the center of actor 1, but this is conducted in the data pipeline (with component `PreNormalize3D`), not in pre-processing. 
3. I do not use sync bn.
4. By default, it's set to False.
5. No, it's implemented in mmcv. `by_epoch=False` provides a smooth lr curve (lr slowly changes iter by iter).

**Comment by JamesMcCullochDickens** [comment]
Thanks.
You can close the issue now if you wish.
